### PR TITLE
Issue #1480: Prevent c8y-config-plugin crashing on bad input

### PR DESCRIPTION
**Pull request by albinsuresh**
## Proposed changes

- [x] Log failures while processing messges and file watcher events instead of propagating the error eventually crashing the plugin
- [ ] Convert errors into SmartREST 502 failure messages

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by initard — COMMENTED** [review]
This error handling prevents the plugin from crashing but does not stop dangling operations (always executing operations), right?

**Comment by albinsuresh** [comment]
> This error handling prevents the plugin from crashing but does not stop dangling operations (always executing operations), right?

Yup. That's the sophisticated error handling that I was talking about (the unchecked tick-box in the PR description), which will be handled in a follow-up PR. This PR is just for damage limitation avoiding crashes on every bad input.

File: plugins/c8y_configuration_plugin/src/main.rs [hunk 1]
```diff
@@ -247,8 +249,7 @@ async fn process_mqtt_message(
                     let config_file_path = config_dir.join(DEFAULT_PLUGIN_CONFIG_FILE);
                     let plugin_config = PluginConfig::new(&config_file_path);
                     // Resend the supported config types
-                    let msg = plugin_config.to_supported_config_types_message()?;
-                    mqtt_client.published.send(msg).await?;
+                    publish_supported_config_types(mqtt_client, &plugin_config).await?;
```
Comment (didier-wenzek) [inline plugins/c8y_configuration_plugin/src/main.rs:252]: Why errors are not catched here? What is the difference between errors coming after an MQTT message compared to an inotify event?

Comment (albinsuresh) [reply plugins/c8y_configuration_plugin/src/main.rs:252]: We're catching the errors one level above, from the caller of this function.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Comment by gligorisaev** [comment]
I am not able to reproduce the crashing

**Pull request merged**